server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -636,10 +642,12 @@ private void stopBeingLeader()
         serviceAnnouncer.unannounce(self);
         metadataRuleManager.stop();
         metadataSegmentManager.stop();
-        leader = false;
+        lookupCoordinatorManager.stop();
       }
       catch (Exception e) {
         log.makeAlert(e, "Unable to stopBeingLeader").emit();
+      } finally {
+        leader = false;
```

Comment:
i see .... i thought it was weird that leader is left to true even in case exceptions ...but you're right that its not related to this PR and may be something for some other day.

reverted.